### DISCUSSION TITLE
Petition for a CONTRIBUTING.md

### DISCUSSION THREAD

**Discussion by ItsIgnacioPortal** [Docile-Alligator/Infinity-For-Reddit#911] [Ideas]
Please consider adding a CONTRIBUTING.md file to the repo, bc as of writing this, it feels like I have to reverse-engineer the repo to know where to even begin adding a new feature.

Here's a how-to:
https://mozillascience.github.io/working-open-workshop/contributing/

**Comment by OHermesJunior** [comment]
I thought of creating one, since I have some knowledge of the codebase already, but I feel like it would be better to restructure the project first, because it is organized by type, not by feature, and it has a lot of random files  on the root folder, which makes it really hard to find where to start. But it would be quite a big change, touching almost all files, which would make it really hard for the dev to accept. I might come back to it though.

**Comment by DanHakimi** [comment]
This would be good. I'll also add -- I"m an attorney (okay, boo me now, we done? kay) and using a CLA or DCO to accept contributions with something like CLA assistant is also good practice. This has not been legal advice and I am not _your_ attorney. Okay, you can get back to booing me now.

**Comment by Docile-Alligator** [comment]
OK. I will check the how-to first. Thank you.

**Comment by Wladefant** [comment]
Hey, right now I am writing the documentation, Readme for the project. Is someone interested in writing the Contributing.md or only the build, installation walk-through?
